### PR TITLE
Form updates for multiplayer SMS games

### LINKED ISSUE #4499: add player names to SMS form for multi-player games
the optimizely test related to #4247 went incredibly well in increasing start and completion rates on SMS and we now want to roll that out for all multi-player games

after consulting with @aaronschachter the best option appears to be:
- default to current form
- if mobilecommons_opt_in_path and mobilecommons_friends_opt_in_path are filled out on https://beta.dosomething.org/admin/config/dosomething/opt_in, keep the form as is
- if ms_game_mp_story_id and sms_game_mp_story_type are filled out on https://beta.dosomething.org/admin/config/dosomething/opt_in then display player names

cc: @ArielSD @tongxiang

### PR DESCRIPTION
**Pull request by DFurnes**
# Step 1: Know It
- This merges in the changes that were A/B tested in #4247. All multiplayer games will now have first name fields. Closes #4499.

For review: @angaither @aaronschachter @tongxiang 
# Step 4: Prove It

For single-player games (configured with `mobilecommons_opt_in_path` and `mobilecommons_friends_opt_in_path`):
screen shot 2015-07-21 at 3 10 12 pm

For multiplayer games (configured with `ms_game_mp_story_id` and `sms_game_mp_story_type`):
screen shot 2015-07-21 at 3 10 22 pm

### CODE REVIEW

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc [hunk 1]
```diff
@@ -183,12 +183,22 @@ function dosomething_signup_sms_game_form($form, &$form_state, $node, $num_frien
...
+    // If this is a multiplayer game, add first name fields.
+    if ($form['sms_game_type']['#default_value'] === 'multi_player') {
+      $form['beta_first_name_' .$i] = array(
+        '#type' => 'textfield',
+        '#title' => 'Your Friend\'s Name',
+        '#required' => TRUE,
+        '#attributes' => array(
+          'data-validate' => 'friend_name',
+          'data-validate-required' => '',
+          'placeholder' => t('(555) 555-5555'),
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc]: mistake placeholder? or am I missing something?

Comment (DFurnes) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc]: That's the grayish placeholder text that shows up on the form. (Or am I missing something else that's wrong haha? :dizzy_face:)

Comment (angaither) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc]: it looks like this is the field for first name, but the placeholder text is a phone number?

Comment (DFurnes) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc]: Oh ho ho. you would be _right_.

Comment (DFurnes) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc]: Fixed!

**Comment by angaither** [comment]
:eyes: good from here! :+1:

**Comment by tongxiang** [comment]
thanks Dave! Looks great. :+1:

**Pull request merged**